### PR TITLE
fix: otp:put requires an authenticated connection

### PR DESCRIPTION
**Pull request by gkc**
**- What I did**
- fix: otp:put requires an authenticated connection
- test: addded unit tests for unauthenticated otp: requests

**- How I did it**
- Moved the auth check in the otp verb handler
- Added missing unit tests.
- Modified the Otp verb in at_server_spec so it also returns true from requiresAuth (for consistency and clarity only, as the otp verb handler has its own check for authentication)
  - updated at_server_spec CHANGELOG
  - updated at_server_spec package version

**- How to verify it**
Tests pass

### CODE REVIEW

**Review by cconstab — APPROVED** [review]

**Comment by gkc** [comment]
@sitaram-kalluri @murali-shris please review as soon as is convenient. If it looks good then please can you also (1) merge this PR to trunk and (2) once the merge actions run has completed, please do a canary release c3.0.44

**Review by sitaram-kalluri — APPROVED** [review]

**Review by murali-shris — APPROVED** [review]

**Pull request merged**
